library/src/main/java/com/climate/mirage/Mirage.java

```diff
@@ -83,9 +89,27 @@
 	private LoadErrorManager loadErrorManager;
 	private ObjectPool<MirageRequest> requestObjectPool;
 	private Context applicationContext;
+    private ActivityLifecycleStub activityLifecycles;
 
 	public Mirage(Context applicationContext) {
 		this.applicationContext = applicationContext.getApplicationContext();
+        ((Application)this.applicationContext).registerActivityLifecycleCallbacks(activityLifecycles = new ActivityLifecycleStub() {
+            @Override
+            public void onActivityDestroyed(Activity activity) {
+                Iterator<Map.Entry<Object, MirageTask>> it = runningRequests.entrySet().iterator();
```

Comment:
Guessing it might be possible of a race condition? onActivityDestroyed callback happens before runningRequests is instantiated?